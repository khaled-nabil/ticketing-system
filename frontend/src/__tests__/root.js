import React from 'react';
import { shallow , mount, render } from 'enzyme';
import Root from '../components/root';

const setup = propOverrides => {
    const props = Object.assign({
        authorized: false
    }, propOverrides);

    const wrapper = render(<Root {...props} />);

    return {
        props,
        wrapper,
        login: wrapper.find("#login-panel").length
    }
};

describe('UI Tests', () => {
    it('Snapshot test', () => {
        const component = shallow(<Root />);
        expect(component).toMatchSnapshot();
    });
    it('Load login screen when no valid token',()=>{
        const component = setup();
        expect(component.login).toEqual(1);
    });
});
