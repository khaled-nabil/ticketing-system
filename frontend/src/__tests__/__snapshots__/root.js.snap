// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Tests Snapshot test 1`] = `ShallowWrapper {}`;
